File: abi-binary-preprocessor/src/test/java/com/memtrip/eosio/Transaction.java
package com.memtrip.eosio;

import com.memtrip.eosio.abi.binary.*;

import java.util.List;

@Abi
public class Transaction {

    private final long expiration;
    private final int refBlockNum;
    private final long refBlockPrefix;
    private final long maxNetUsageWords;
    private final long maxCpuUsageMs;
    private final long delaySec;
    private final List<PackedAction> contextFreeActions;
    private final List<PackedAction> actions;
    private final List<String> transactionExtensions;
    private final List<String> signatures;
    private final List<String> contextFreeData;

    @TimestampCompress
    public long getExpiration() {
        return expiration;
    }

    @BlockNumCompress
    public int getRefBlockNum() {
        return refBlockNum;
    }

    @BlockPrefixCompress
    public long getRefBlockPrefix() {
        return refBlockPrefix;
    }

    @VariableUIntCompress
    public long getMaxNetUsageWords() {
        return maxNetUsageWords;
    }

    @VariableUIntCompress
    public long getMaxCpuUsageMs() {
        return maxCpuUsageMs;
    }

    @VariableUIntCompress
    public long getDelaySec() {
        return delaySec;
    }

    @CollectionCompress
    public List<PackedAction> getContextFreeActions() {
        return contextFreeActions;
    }

    @CollectionCompress
    public List<PackedAction> getActions() {
        return actions;
    }

    @StringCollectionCompress
    public List<String> getTransactionExtensions() {
        return transactionExtensions;
    }

    public List<String> getSignatures() {
        return signatures;
    }

    public List<String> getContextFreeData() {
        return contextFreeData;
    }

    public Transaction(
        long expiration,
        int refBlockNum,
        long refBlockPrefix,
        long maxNetUsageWords,
        long maxCpuUsageMs,
        long delaySec,
        List<PackedAction> contextFreeActions,
        List<PackedAction> actions,
        List<String> transactionExtensions,
        List<String> signatures,
        List<String> contextFreeData
    ) {
        this.expiration = expiration;
        this.refBlockNum = refBlockNum;
        this.refBlockPrefix = refBlockPrefix;
        this.maxNetUsageWords = maxNetUsageWords;
        this.maxCpuUsageMs = maxCpuUsageMs;
        this.delaySec = delaySec;
        this.contextFreeActions = contextFreeActions;
        this.actions = actions;
        this.transactionExtensions = transactionExtensions;
        this.signatures = signatures;
        this.contextFreeData = contextFreeData;
    }
}

File: eos-http-rpc/app/src/main/java/com/memtrip/eos/http/rpc/model/transaction/request/Transaction.kt
package com.memtrip.eos.http.rpc.model.transaction.request

import com.memtrip.eos.abi.writer.Abi
import com.memtrip.eos.abi.writer.BlockNumCompress
import com.memtrip.eos.abi.writer.BlockPrefixCompress
import com.memtrip.eos.abi.writer.CollectionCompress
import com.memtrip.eos.abi.writer.StringCollectionCompress
import com.memtrip.eos.abi.writer.TimestampCompress
import com.memtrip.eos.abi.writer.VariableUIntCompress
import com.squareup.moshi.JsonClass
import org.threeten.bp.LocalDateTime
import org.threeten.bp.ZoneOffset

@Abi
@JsonClass(generateAdapter = true)
class Transaction(
    val expiration: LocalDateTime,
    val ref_block_num: Int,
    val ref_block_prefix: Long,
    val max_net_usage_words: Long,
    val max_cpu_usage_ms: Long,
    val delay_sec: Long,
    val context_free_actions: List<Action>,
    val actions: List<Action>,
    val transaction_extensions: List<String>,
    val signatures: List<String>,
    val context_free_data: List<String>) {

    val getExpiration: Long
        @TimestampCompress get() = expiration.toInstant(ZoneOffset.UTC).toEpochMilli()

    val getRefBlockNum: Int
        @BlockNumCompress get() = ref_block_num

    val getRefBlockPrefix: Long
        @BlockPrefixCompress get() = ref_block_prefix

    val getMaxNetUsageWords: Long
        @VariableUIntCompress get() = max_net_usage_words

    val getMaxCpuUsageMs: Long
        @VariableUIntCompress get() = max_cpu_usage_ms

    val getDelaySec: Long
        @VariableUIntCompress get() = delay_sec

    val getContextFreeActions: List<Action>
        @CollectionCompress get() = context_free_actions

    val getActions: List<Action>
        @CollectionCompress get() = actions

    val getTransactionExtensions: List<String>
        @StringCollectionCompress get() = transaction_extensions
}